Update LFPy for LFPykit-0.5.dev; start LFPy-2.3.dev

**Describe the bug**
Starting LFPykit-0.5.dev  and fixing https://github.com/LFPy/LFPykit/issues/160 now results in broken LFPy tests: 
```
FAILED LFPy/test/test_lfpcalc.py::testLfpCalc::test_lfpcalc_calc_lfp_linesource_00 - TypeError: calc_lfp_linesource() missing 2 required positional arguments: 'cell_y' and 'cell_z'
...
```
LFPy-2.3 should in the future depend on LFPykit-0.5; so for LFPy-2.2.* we should require LFPykit-0.4.*, as more LFPykit updates may further break LFPy.

--- a/LFPy/test/test_lfpcalc.py
+++ b/LFPy/test/test_lfpcalc.py
@@ -55,7 +55,7 @@ class testLfpCalc(unittest.TestCase):
                                     x=0.5, y=0, z=1,
                                     sigma=sigma,
                                     r_limit=cell.d / 2
-                                ))
+        ))
 
     def test_lfpcalc_calc_lfp_pointsource_moi_00(self):
         """
@@ -316,14 +316,19 @@ class testLfpCalc(unittest.TestCase):
                                                        h=h,
                                                        steps=steps)
 
-        without_saline = lfpcalc.calc_lfp_pointsource_moi(cell.x, cell.y, cell.z,
-                                                          x=0, y=0, z=0,
-                                                          sigma_T=sigma_T,
-                                                          sigma_G=sigma_G,
-                                                          sigma_S=sigma_T,
-                                                          r_limit=cell.d / 2,
-                                                          h=h,
-                                                          steps=steps)
+        without_saline = lfpcalc.calc_lfp_pointsource_moi(
+            cell.x,
+            cell.y,
+            cell.z,
+            x=0,
+            y=0,
+            z=0,
+            sigma_T=sigma_T,
+            sigma_G=sigma_G,
+            sigma_S=sigma_T,
+            r_limit=cell.d / 2,
+            h=h,
+            steps=steps)
 
         np.testing.assert_array_less(with_saline, without_saline)
 
@@ -348,14 +353,19 @@ class testLfpCalc(unittest.TestCase):
                                                       h=h,
                                                       steps=steps)
 
-        without_saline = lfpcalc.calc_lfp_linesource_moi(cell.x, cell.y, cell.z,
-                                                         x=0, y=0, z=0,
-                                                         sigma_T=sigma_T,
-                                                         sigma_G=sigma_G,
-                                                         sigma_S=sigma_T,
-                                                         r_limit=cell.d / 2,
-                                                         h=h,
-                                                         steps=steps)
+        without_saline = lfpcalc.calc_lfp_linesource_moi(
+            cell.x,
+            cell.y,
+            cell.z,
+            x=0,
+            y=0,
+            z=0,
+            sigma_T=sigma_T,
+            sigma_G=sigma_G,
+            sigma_S=sigma_T,
+            r_limit=cell.d / 2,
+            h=h,
+            steps=steps)
 
         np.testing.assert_array_less(with_saline, without_saline)
 
@@ -371,23 +381,33 @@ class testLfpCalc(unittest.TestCase):
 
         cell = DummyCell(z=np.array([[100, 100]]))
 
-        with_saline = lfpcalc.calc_lfp_root_as_point_moi(cell.x, cell.y, cell.z,
-                                                         x=0, y=0, z=0,
-                                                         sigma_T=sigma_T,
-                                                         sigma_G=sigma_G,
-                                                         sigma_S=sigma_S,
-                                                         r_limit=cell.d / 2,
-                                                         h=h,
-                                                         steps=steps)
+        with_saline = lfpcalc.calc_lfp_root_as_point_moi(
+            cell.x,
+            cell.y,
+            cell.z,
+            x=0,
+            y=0,
+            z=0,
+            sigma_T=sigma_T,
+            sigma_G=sigma_G,
+            sigma_S=sigma_S,
+            r_limit=cell.d / 2,
+            h=h,
+            steps=steps)
 
-        without_saline = lfpcalc.calc_lfp_root_as_point_moi(cell.x, cell.y, cell.z,
-                                                            x=0, y=0, z=0,
-                                                            sigma_T=sigma_T,
-                                                            sigma_G=sigma_G,
-                                                            sigma_S=sigma_T,
-                                                            r_limit=cell.d / 2,
-                                                            h=h,
-                                                            steps=steps)
+        without_saline = lfpcalc.calc_lfp_root_as_point_moi(
+            cell.x,
+            cell.y,
+            cell.z,
+            x=0,
+            y=0,
+            z=0,
+            sigma_T=sigma_T,
+            sigma_G=sigma_G,
+            sigma_S=sigma_T,
+            r_limit=cell.d / 2,
+            h=h,
+            steps=steps)
 
         np.testing.assert_array_less(with_saline, without_saline)
 
@@ -493,14 +513,19 @@ class testLfpCalc(unittest.TestCase):
                                                        h=h,
                                                        steps=steps)
 
-        without_saline = lfpcalc.calc_lfp_pointsource_moi(cell.x, cell.y, cell.z,
-                                                          x=0, y=0, z=50,
-                                                          sigma_T=sigma_T,
-                                                          sigma_G=sigma_G,
-                                                          sigma_S=sigma_T,
-                                                          r_limit=cell.d / 2,
-                                                          h=h,
-                                                          steps=steps)
+        without_saline = lfpcalc.calc_lfp_pointsource_moi(
+            cell.x,
+            cell.y,
+            cell.z,
+            x=0,
+            y=0,
+            z=50,
+            sigma_T=sigma_T,
+            sigma_G=sigma_G,
+            sigma_S=sigma_T,
+            r_limit=cell.d / 2,
+            h=h,
+            steps=steps)
 
         np.testing.assert_almost_equal(with_saline, without_saline)
 
@@ -525,14 +550,19 @@ class testLfpCalc(unittest.TestCase):
                                                       h=h,
                                                       steps=steps)
 
-        without_saline = lfpcalc.calc_lfp_linesource_moi(cell.x, cell.y, cell.z,
-                                                         x=0, y=0, z=0,
-                                                         sigma_T=sigma_T,
-                                                         sigma_G=sigma_G,
-                                                         sigma_S=sigma_T,
-                                                         r_limit=cell.d / 2,
-                                                         h=h,
-                                                         steps=steps)
+        without_saline = lfpcalc.calc_lfp_linesource_moi(
+            cell.x,
+            cell.y,
+            cell.z,
+            x=0,
+            y=0,
+            z=0,
+            sigma_T=sigma_T,
+            sigma_G=sigma_G,
+            sigma_S=sigma_T,
+            r_limit=cell.d / 2,
+            h=h,
+            steps=steps)
 
         np.testing.assert_almost_equal(with_saline, without_saline)
 
@@ -548,23 +578,33 @@ class testLfpCalc(unittest.TestCase):
 
         cell = DummyCell(z=np.array([[100, 100]]))
 
-        with_saline = lfpcalc.calc_lfp_root_as_point_moi(cell.x, cell.y, cell.z,
-                                                         x=0, y=0, z=0,
-                                                         sigma_T=sigma_T,
-                                                         sigma_G=sigma_G,
-                                                         sigma_S=sigma_S,
-                                                         r_limit=cell.d / 2,
-                                                         h=h,
-                                                         steps=steps)
+        with_saline = lfpcalc.calc_lfp_root_as_point_moi(
+            cell.x,
+            cell.y,
+            cell.z,
+            x=0,
+            y=0,
+            z=0,
+            sigma_T=sigma_T,
+            sigma_G=sigma_G,
+            sigma_S=sigma_S,
+            r_limit=cell.d / 2,
+            h=h,
+            steps=steps)
 
-        without_saline = lfpcalc.calc_lfp_root_as_point_moi(cell.x, cell.y, cell.z,
-                                                            x=0, y=0, z=0,
-                                                            sigma_T=sigma_T,
-                                                            sigma_G=sigma_G,
-                                                            sigma_S=sigma_T,
-                                                            r_limit=cell.d / 2,
-                                                            h=h,
-                                                            steps=steps)
+        without_saline = lfpcalc.calc_lfp_root_as_point_moi(
+            cell.x,
+            cell.y,
+            cell.z,
+            x=0,
+            y=0,
+            z=0,
+            sigma_T=sigma_T,
+            sigma_G=sigma_G,
+            sigma_S=sigma_T,
+            r_limit=cell.d / 2,
+            h=h,
+            steps=steps)
 
         np.testing.assert_almost_equal(with_saline, without_saline)
 

--- a/LFPy/test/test_lfpcalc.py
+++ b/LFPy/test/test_lfpcalc.py
@@ -50,11 +50,12 @@ class testLfpCalc(unittest.TestCase):
         sigma = 0.3
         cell = DummyCell()
         np.testing.assert_equal(1. / (4 * np.pi * sigma),
-                                lfpcalc.calc_lfp_pointsource(cell,
-                                                             x=0.5, y=0, z=1,
-                                                             sigma=sigma,
-                                                             r_limit=cell.d / 2
-                                                             ))
+                                lfpcalc.calc_lfp_pointsource(
+                                    cell.x, cell.y, cell.z,
+                                    x=0.5, y=0, z=1,
+                                    sigma=sigma,
+                                    r_limit=cell.d / 2
+                                ))
 
     def test_lfpcalc_calc_lfp_pointsource_moi_00(self):
         """
@@ -68,10 +69,10 @@ class testLfpCalc(unittest.TestCase):
         steps = 20
         cell = DummyCell(np.array([[h / 2, h / 2]]))
 
-        in_vivo = lfpcalc.calc_lfp_pointsource(cell,
+        in_vivo = lfpcalc.calc_lfp_pointsource(cell.x, cell.y, cell.z,
                                                x=0.5, y=0, z=1, sigma=sigma_T,
                                                r_limit=cell.d / 2)
-        in_vitro = lfpcalc.calc_lfp_pointsource_moi(cell,
+        in_vitro = lfpcalc.calc_lfp_pointsource_moi(cell.x, cell.y, cell.z,
                                                     x=0.5, y=0, z=1,
                                                     sigma_T=sigma_T,
                                                     sigma_G=sigma_G,
@@ -105,7 +106,9 @@ class testLfpCalc(unittest.TestCase):
             source_scaling / np.abs(z - (2 * h - cell.z.mean(axis=-1)[0]))
         )
 
-        moi_method_lfpy = lfpcalc.calc_lfp_pointsource_moi(cell,
+        moi_method_lfpy = lfpcalc.calc_lfp_pointsource_moi(cell.x,
+                                                           cell.y,
+                                                           cell.z,
                                                            x=0., y=0, z=z,
                                                            sigma_T=sigma_T,
                                                            sigma_G=sigma_G,
@@ -127,11 +130,11 @@ class testLfpCalc(unittest.TestCase):
         steps = 20
         cell = DummyCell(z=np.array([[h / 2, h / 2]]))
 
-        in_vivo = lfpcalc.calc_lfp_pointsource(cell,
+        in_vivo = lfpcalc.calc_lfp_pointsource(cell.x, cell.y, cell.z,
                                                x=0.5, y=0, z=h / 2,
                                                sigma=sigma_T,
                                                r_limit=cell.d / 2)
-        in_vitro = lfpcalc.calc_lfp_pointsource_moi(cell,
+        in_vitro = lfpcalc.calc_lfp_pointsource_moi(cell.x, cell.y, cell.z,
                                                     x=0.5, y=0, z=h / 2,
                                                     sigma_T=sigma_T,
                                                     sigma_G=sigma_G,
@@ -155,10 +158,10 @@ class testLfpCalc(unittest.TestCase):
         steps = 20
         cell = DummyCell()
 
-        in_vivo = lfpcalc.calc_lfp_linesource(cell,
+        in_vivo = lfpcalc.calc_lfp_linesource(cell.x, cell.y, cell.z,
                                               x=0.5, y=0, z=0, sigma=sigma_T,
                                               r_limit=cell.d / 2)
-        in_vitro = lfpcalc.calc_lfp_linesource_moi(cell,
+        in_vitro = lfpcalc.calc_lfp_linesource_moi(cell.x, cell.y, cell.z,
                                                    x=0.5, y=0, z=0,
                                                    sigma_T=sigma_T,
                                                    sigma_G=sigma_G,
@@ -181,11 +184,11 @@ class testLfpCalc(unittest.TestCase):
         steps = 20
         cell = DummyCell()
 
-        in_vivo = lfpcalc.calc_lfp_root_as_point(cell,
+        in_vivo = lfpcalc.calc_lfp_root_as_point(cell.x, cell.y, cell.z,
                                                  x=0.0, y=0, z=0,
                                                  sigma=sigma_T,
                                                  r_limit=cell.d / 2)
-        in_vitro = lfpcalc.calc_lfp_root_as_point_moi(cell,
+        in_vitro = lfpcalc.calc_lfp_root_as_point_moi(cell.x, cell.y, cell.z,
                                                       x=0.0, y=0, z=0,
                                                       sigma_T=sigma_T,
                                                       sigma_G=sigma_G,
@@ -203,7 +206,7 @@ class testLfpCalc(unittest.TestCase):
         sigma_T = 0.3
         cell = DummyCell()
 
-        in_vivo = lfpcalc.calc_lfp_linesource(cell,
+        in_vivo = lfpcalc.calc_lfp_linesource(cell.x, cell.y, cell.z,
                                               x=0.5, y=0.0, z=0, sigma=sigma_T,
                                               r_limit=cell.d / 2)[0]
         np.testing.assert_array_less(in_vivo, 1e12)
@@ -221,11 +224,11 @@ class testLfpCalc(unittest.TestCase):
 
         cell = DummyCell(z=np.array([[50, 50]]))
 
-        in_vivo = lfpcalc.calc_lfp_pointsource(cell,
+        in_vivo = lfpcalc.calc_lfp_pointsource(cell.x, cell.y, cell.z,
                                                x=50., y=0, z=0,
                                                sigma=sigma_T,
                                                r_limit=cell.d / 2)
-        in_vitro = lfpcalc.calc_lfp_pointsource_moi(cell,
+        in_vitro = lfpcalc.calc_lfp_pointsource_moi(cell.x, cell.y, cell.z,
                                                     x=50, y=0, z=0,
                                                     sigma_T=sigma_T,
                                                     sigma_G=sigma_G,
@@ -249,11 +252,11 @@ class testLfpCalc(unittest.TestCase):
 
         cell = DummyCell(z=np.array([[50, 50]]))
 
-        in_vivo = lfpcalc.calc_lfp_linesource(cell,
+        in_vivo = lfpcalc.calc_lfp_linesource(cell.x, cell.y, cell.z,
                                               x=50., y=0, z=0,
                                               sigma=sigma_T,
                                               r_limit=cell.d / 2)
-        in_vitro = lfpcalc.calc_lfp_linesource_moi(cell,
+        in_vitro = lfpcalc.calc_lfp_linesource_moi(cell.x, cell.y, cell.z,
                                                    x=50, y=0, z=0,
                                                    sigma_T=sigma_T,
                                                    sigma_G=sigma_G,
@@ -277,11 +280,11 @@ class testLfpCalc(unittest.TestCase):
 
         cell = DummyCell(z=np.array([[50, 50]]))
 
-        in_vivo = lfpcalc.calc_lfp_root_as_point(cell,
+        in_vivo = lfpcalc.calc_lfp_root_as_point(cell.x, cell.y, cell.z,
                                                  x=50., y=0, z=0,
                                                  sigma=sigma_T,
                                                  r_limit=cell.d / 2)
-        in_vitro = lfpcalc.calc_lfp_root_as_point_moi(cell,
+        in_vitro = lfpcalc.calc_lfp_root_as_point_moi(cell.x, cell.y, cell.z,
                                                       x=50, y=0, z=0,
                                                       sigma_T=sigma_T,
                                                       sigma_G=sigma_G,
@@ -304,7 +307,7 @@ class testLfpCalc(unittest.TestCase):
 
         cell = DummyCell(z=np.array([[100, 100]]))
 
-        with_saline = lfpcalc.calc_lfp_pointsource_moi(cell,
+        with_saline = lfpcalc.calc_lfp_pointsource_moi(cell.x, cell.y, cell.z,
                                                        x=0, y=0, z=0,
                                                        sigma_T=sigma_T,
                                                        sigma_G=sigma_G,
@@ -313,7 +316,7 @@ class testLfpCalc(unittest.TestCase):
                                                        h=h,
                                                        steps=steps)
 
-        without_saline = lfpcalc.calc_lfp_pointsource_moi(cell,
+        without_saline = lfpcalc.calc_lfp_pointsource_moi(cell.x, cell.y, cell.z,
                                                           x=0, y=0, z=0,
                                                           sigma_T=sigma_T,
                                                           sigma_G=sigma_G,
@@ -336,7 +339,7 @@ class testLfpCalc(unittest.TestCase):
 
         cell = DummyCell(z=np.array([[100, 100]]))
 
-        with_saline = lfpcalc.calc_lfp_linesource_moi(cell,
+        with_saline = lfpcalc.calc_lfp_linesource_moi(cell.x, cell.y, cell.z,
                                                       x=0, y=0, z=0,
                                                       sigma_T=sigma_T,
                                                       sigma_G=sigma_G,
@@ -345,7 +348,7 @@ class testLfpCalc(unittest.TestCase):
                                                       h=h,
                                                       steps=steps)
 
-        without_saline = lfpcalc.calc_lfp_linesource_moi(cell,
+        without_saline = lfpcalc.calc_lfp_linesource_moi(cell.x, cell.y, cell.z,
                                                          x=0, y=0, z=0,
                                                          sigma_T=sigma_T,
                                                          sigma_G=sigma_G,
@@ -368,7 +371,7 @@ class testLfpCalc(unittest.TestCase):
 
         cell = DummyCell(z=np.array([[100, 100]]))
 
-        with_saline = lfpcalc.calc_lfp_root_as_point_moi(cell,
+        with_saline = lfpcalc.calc_lfp_root_as_point_moi(cell.x, cell.y, cell.z,
                                                          x=0, y=0, z=0,
                                                          sigma_T=sigma_T,
                                                          sigma_G=sigma_G,
@@ -377,7 +380,7 @@ class testLfpCalc(unittest.TestCase):
                                                          h=h,
                                                          steps=steps)
 
-        without_saline = lfpcalc.calc_lfp_root_as_point_moi(cell,
+        without_saline = lfpcalc.calc_lfp_root_as_point_moi(cell.x, cell.y, cell.z,
                                                             x=0, y=0, z=0,
                                                             sigma_T=sigma_T,
                                                             sigma_G=sigma_G,
@@ -403,7 +406,7 @@ class testLfpCalc(unittest.TestCase):
 
         cell = DummyCell(x=np.array([[0., -200.]]), z=np.array(([[0., 220.]])))
 
-        calculated = lfpcalc.calc_lfp_pointsource_moi(cell,
+        calculated = lfpcalc.calc_lfp_pointsource_moi(cell.x, cell.y, cell.z,
                                                       x=100, y=0, z=0,
                                                       sigma_T=sigma_T,
                                                       sigma_G=sigma_G,
@@ -430,7 +433,7 @@ class testLfpCalc(unittest.TestCase):
         cell = DummyCell(x=np.array([[-100, 50]]),
                          z=np.array([[0, 110]]))
 
-        calculated = lfpcalc.calc_lfp_linesource_moi(cell,
+        calculated = lfpcalc.calc_lfp_linesource_moi(cell.x, cell.y, cell.z,
                                                      x=100, y=0, z=0,
                                                      sigma_T=sigma_T,
                                                      sigma_G=sigma_G,
@@ -458,7 +461,7 @@ class testLfpCalc(unittest.TestCase):
                          y=np.array([[0., 0.]]),
                          z=np.array([[0, 220]]))
 
-        calculated = lfpcalc.calc_lfp_root_as_point_moi(cell,
+        calculated = lfpcalc.calc_lfp_root_as_point_moi(cell.x, cell.y, cell.z,
                                                         x=100, y=0, z=0,
                                                         sigma_T=sigma_T,
                                                         sigma_G=sigma_G,
@@ -481,7 +484,7 @@ class testLfpCalc(unittest.TestCase):
 
         cell = DummyCell(z=np.array([[100, 100]]))
 
-        with_saline = lfpcalc.calc_lfp_pointsource_moi(cell,
+        with_saline = lfpcalc.calc_lfp_pointsource_moi(cell.x, cell.y, cell.z,
                                                        x=0, y=0, z=50,
                                                        sigma_T=sigma_T,
                                                        sigma_G=sigma_G,
@@ -490,7 +493,7 @@ class testLfpCalc(unittest.TestCase):
                                                        h=h,
                                                        steps=steps)
 
-        without_saline = lfpcalc.calc_lfp_pointsource_moi(cell,
+        without_saline = lfpcalc.calc_lfp_pointsource_moi(cell.x, cell.y, cell.z,
                                                           x=0, y=0, z=50,
                                                           sigma_T=sigma_T,
                                                           sigma_G=sigma_G,
@@ -513,7 +516,7 @@ class testLfpCalc(unittest.TestCase):
 
         cell = DummyCell(z=np.array([[100, 100]]))
 
-        with_saline = lfpcalc.calc_lfp_linesource_moi(cell,
+        with_saline = lfpcalc.calc_lfp_linesource_moi(cell.x, cell.y, cell.z,
                                                       x=0, y=0, z=0,
                                                       sigma_T=sigma_T,
                                                       sigma_G=sigma_G,
@@ -522,7 +525,7 @@ class testLfpCalc(unittest.TestCase):
                                                       h=h,
                                                       steps=steps)
 
-        without_saline = lfpcalc.calc_lfp_linesource_moi(cell,
+        without_saline = lfpcalc.calc_lfp_linesource_moi(cell.x, cell.y, cell.z,
                                                          x=0, y=0, z=0,
                                                          sigma_T=sigma_T,
                                                          sigma_G=sigma_G,
@@ -545,7 +548,7 @@ class testLfpCalc(unittest.TestCase):
 
         cell = DummyCell(z=np.array([[100, 100]]))
 
-        with_saline = lfpcalc.calc_lfp_root_as_point_moi(cell,
+        with_saline = lfpcalc.calc_lfp_root_as_point_moi(cell.x, cell.y, cell.z,
                                                          x=0, y=0, z=0,
                                                          sigma_T=sigma_T,
                                                          sigma_G=sigma_G,
@@ -554,7 +557,7 @@ class testLfpCalc(unittest.TestCase):
                                                          h=h,
                                                          steps=steps)
 
-        without_saline = lfpcalc.calc_lfp_root_as_point_moi(cell,
+        without_saline = lfpcalc.calc_lfp_root_as_point_moi(cell.x, cell.y, cell.z,
                                                             x=0, y=0, z=0,
                                                             sigma_T=sigma_T,
                                                             sigma_G=sigma_G,
@@ -580,7 +583,8 @@ class testLfpCalc(unittest.TestCase):
         phi_analytic = 1. / (4 * np.pi * sigma_r)
         np.testing.assert_equal(
             phi_analytic, lfpcalc.calc_lfp_pointsource_anisotropic(
-                cell, x=0, y=0, z=0, sigma=sigma, r_limit=cell.d / 2))
+                cell.x, cell.y, cell.z, x=0, y=0, z=0, sigma=sigma,
+                r_limit=cell.d / 2))
 
     def test_deltaS_calc(self):
         cell = DummyCell(y=np.array([[0, 5]]))

--- a/LFPy/version.py
+++ b/LFPy/version.py
@@ -1,1 +1,1 @@
-version = "2.2.5"
+version = "2.3.dev"

--- a/requirements.txt
+++ b/requirements.txt
@@ -7,4 +7,5 @@ h5py>=2.5
 mpi4py>=1.2
 matplotlib>=2.0
 neuron>=7.7.2
-LFPykit>=0.4,<0.5
+# LFPykit>=0.5.dev
+git+https://github.com/LFPy/LFPykit@master#egg=lfpykit

--- a/setup.py
+++ b/setup.py
@@ -107,7 +107,7 @@ setup(
         'Cython>=0.20',
         'h5py>=2.5',
         'mpi4py>=1.2',
-        'LFPykit>=0.4,<0.5'
+        'lfpykit@git+https://github.com/LFPy/LFPykit.git#egg=lfpykit'
     ],
     extras_require={'tests': ['pytest'],
                     'docs': ['sphinx', 'numpydoc', 'sphinx_rtd_theme']

